In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [9]:
byte_level[(byte_level['Global model parameters'] <= 1.3e8) & (byte_level['Global model parameters'] >= 1.1e8) & (byte_level['Global model layers'] >= 15)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
5249,s,4.0,1.050125e+10,3825664.00,1.228559e+08,11476992.00,832,640,15,3
5749,s,4.0,1.042370e+10,3098787.84,1.228559e+08,12395151.36,832,576,15,4
5809,s,4.0,1.198473e+10,2448424.96,1.117657e+08,9793699.84,768,512,16,4
5813,s,4.0,1.092582e+10,3098787.84,1.117657e+08,12395151.36,768,576,16,4
5886,s,4.0,1.153588e+10,2448424.96,1.188495e+08,9793699.84,768,512,17,4
5890,s,4.0,1.055154e+10,3098787.84,1.188495e+08,12395151.36,768,576,17,4
6016,s,4.0,1.187885e+10,2448424.96,1.117713e+08,9793699.84,704,512,19,4
6020,s,4.0,1.083775e+10,3098787.84,1.117713e+08,12395151.36,704,576,19,4
6081,s,4.0,1.150003e+10,2448424.96,1.164004e+08,9793699.84,704,512,20,4
6085,s,4.0,1.052154e+10,3098787.84,1.164004e+08,12395151.36,704,576,20,4


In [3]:
byte_level[(byte_level['Tokens'] >= 4.8e10)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
0,s,2,6.312902e+10,344309.76,7498301.44,688619.52,448,192,4,2
118,s,2,5.509120e+10,344309.76,8263434.24,688619.52,384,192,6,2
264,s,2,5.450177e+10,344309.76,7651328.00,688619.52,320,192,8,2
334,s,2,4.937376e+10,344309.76,8607744.00,688619.52,320,192,9,2
1089,s,2,5.266842e+10,153026.56,7345274.88,612106.24,256,128,12,4
1441,s,2,5.133669e+10,153026.56,7345274.88,765132.80,256,128,12,5


4e18 FLOPs
===

Patch size = 2

1102: 4.9e10 tokens, 11M global model, 1.2M decoder, 384 global model dimension, 256 encoder/decoder dimension, 6 global model layers, 2 decoder layers

1421: 2.2e+10 tokens, 28M global model, 3M decoder, 512	global model dimension, 320	encoder/decoder dimension, 9 global model layers, 3 decoder layers

1487: 1.2e+10 tokens, 58M global model,	6M decoder, 704	global model dimension, 448	encoder/decoder dimension, 10 global model layers, 3 decoder layers

Patch size = 4

6e10 tokens, 19M global model,	1.8M decoder, 448 global model dimension, 256 encoder/decoder dimension, 8 global model layers,	3 decoder layers

2e+10 tokens, 59M global model,	5.5M decoder, 576 global model dimension, 384 encoder/decoder dimension, 15	global model layer, 4 decoder layer

1.2e+10	tokens, 120M global model, 9.8M decoder, 768 global model dimension, 512 encoder/decoder dimension, 17 global model layer, 4 decoder layer